### DISCUSSION TITLE
Add-PnPGroupMember not working for "new guests"

### DISCUSSION THREAD

**Discussion by sajense** [pnp/powershell#3468] [Q&A]
I'm having an issue getting the cmdlet to work when used with the app registration. I can add existing users and existing guests, but when i try to add a new guest (user does not yet exist in my tenant) it "fails". No error is returned and the tracelog is empty, but nothing is happening.
It works correct when i use the command in my own user context, so i assumed it was a permission issue at first, but i've tried granting it all permissions i can think off, and it still does not work.

Is this expected or am i missing something totally obvious? 😆

**Comment by AshokChava** [comment]
@sajense did you find any solution for this? I am also facing the exact same issue.

**Reply by sajense** [reply]
@AshokChava Good question .. been so long I barely forgot. I think I managed to solve it by assigning both the graph api permission User.Invite.All AND also grant the service principal the Entra Id role Guest Inviter.

**Comment by AshokChava** [comment]
hmm still not able to make it work I don't see any mails going to the external user and don't see any error as well.
This is my code

foreach ($member in $visitors) {
    $user=Get-PnPUser | ? Email -eq $member
    if ($user -eq $null)
    {
        Write-Host "Non-existing user sending out a mail $member"
        $res=Add-PnPGroupMember -EmailAddress $member -Group 4 -SendEmail
    }
    else
    {
        Write-Host "Existing user adding to the site"
        Add-PnPGroupMember  -LoginName $member -Group 5
    }
}

**Comment by sajense** [comment]
Hmm seems like i didn't either. I found my old code. I was building a function app that served a webpage for users to do some stuff on their sharepoint site. I ended up ditching the approach and going with some power automate flows and microsoft forms integrated in their microsoft team instead.

This is what i ended up doing in the old code as a workaround

```
# Check email addresses exist in tenant
foreach($email in $emailAddresses){
    try {
        $user = Get-MgUser -Filter "Mail eq '$email'"
        if ($null -eq $user) {
            New-MgInvitation -InvitedUserEmailAddress $email -InviteRedirectUrl "https://myapplications.microsoft.com" -SendInvitationMessage:$false
        }
    }
    catch {
        $responseMessage += "$_"
    }
}

# Loop until users return on get
foreach($email in $emailAddresses){
    do {
        Start-Sleep -Seconds 7
    } until (
        $null -ne (Get-MgUser -Filter "Mail eq '$email'")
    )
}
Start-Sleep -Seconds 20

foreach($email in $emailAddresses){
    try {
        Add-PnPGroupMember -Group $projectGroupName -EmailAddress $email -Connection $conn5
        $responseMessage += "$($email) was added to the group: '$projectGroupName'."
    }
    catch {
        $responseMessage += "$_"
    }
}
```
